### DISCUSSION TITLE
Bittorrent via VPN with incoming ports?

### DISCUSSION THREAD

**Discussion by bogorad** [XTLS/Xray-core#2653] [Q&A] [answered]
I have a commercial VPN client set up. The VPN allows incoming ports. Right now, I set everything up inside my OpenWRT: it just forwards the specified port from `WAN` firewall zone to a machine in my LAN:

![image](https://github.com/XTLS/Xray-core/assets/2777398/a090b527-997b-4ca2-a6f9-e3e2b76b3e1e)

I tried moving the wg-client to XTLS. The setup is easy; forwarding all traffic via NFTables to `Dokodemo-Door` is easy. But how do I address the open port on the VPN side so I can forward it back to my torrent machine?

**Comment by us254** [comment, marked as answer]
```
"inbounds": [{
  "port": "Your_VPN_Port",
  "protocol": "dokodemo-door",
  "settings": {
    "network": "tcp,udp",
    "followRedirect": true,
    "address": "address_of_torrent_machine"
  }
}]
```
First, you need to ensure that your VPN provider supports port forwarding. you need to configure Xray to forward traffic from this port to your torrent machine. You will need to specify the open VPN port, protocol (TCP/UDP), and the address of your torrent machine in the Dokodemo-Door settings in your Xray configuration file. ensure that the networking on your OpenWRT is set up correctly to route traffic from the VPN to the Xray instance, and consequently, to the torrent machine.
